From ed5a05477a5931dc8eb57d20c916928cb8024f21 Mon Sep 17 00:00:00 2001
Message-Id: <ed5a05477a5931dc8eb57d20c916928cb8024f21.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: =?UTF-8?q?Ville=20Syrj=C3=A4l=C3=A4?= <ville.syrjala@linux.intel.com>
Date: Sat, 28 Jun 2014 02:04:06 +0300
Subject: [PATCH 233/312] FROM_UPSTREAM [VPG]: drm/i915: Clear TX FIFO reset
 master override bits on chv
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Clear the override bits to make sure the hardware maanages
the TX FIFO reset master on its own.

Change-Id: I50d8d1da9570c491f2d8fe2f3d92a78fd34af600
Signed-off-by: Ville Syrjälä <ville.syrjala@linux.intel.com>
Signed-off-by: Deepak S <deepak.s@intel.com>
Upstream-repo: http://patchwork.freedesktop.org/patch/28732/
Upstream-commit: <not applicable>
Upstream-as-of-tag: <not applicable>
---
 drivers/gpu/drm/i915/intel_dp.c   |   20 +++-----------------
 drivers/gpu/drm/i915/intel_hdmi.c |   20 +++-----------------
 2 files changed, 6 insertions(+), 34 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_dp.c b/drivers/gpu/drm/i915/intel_dp.c
index 429bd58..8d43516 100644
--- a/drivers/gpu/drm/i915/intel_dp.c
+++ b/drivers/gpu/drm/i915/intel_dp.c
@@ -2261,27 +2261,13 @@ static void chv_pre_enable_dp(struct intel_encoder *encoder)
 
 	mutex_lock(&dev_priv->dpio_lock);
 
-	/* TX FIFO reset source */
-	val = vlv_dpio_read(dev_priv, pipe, VLV_PCS01_DW0(ch));
-	val |= DPIO_LEFT_TXFIFO_RST_MASTER2;
-	val &= ~DPIO_LEFT_TXFIFO_RST_MASTER2;
-	vlv_dpio_write(dev_priv, pipe, VLV_PCS01_DW0(ch), val);
-
+	/* allow hardware to manage TX FIFO reset source */
 	val = vlv_dpio_read(dev_priv, pipe, VLV_PCS01_DW11(ch));
-	val &= ~DPIO_LEFT_TXFIFO_RST_MASTER;
-	val &= ~DPIO_RIGHT_TXFIFO_RST_MASTER;
-	val |= DPIO_LANEDESKEW_STRAP_OVRD;
+	val &= ~DPIO_LANEDESKEW_STRAP_OVRD;
 	vlv_dpio_write(dev_priv, pipe, VLV_PCS01_DW11(ch), val);
 
-	val = vlv_dpio_read(dev_priv, pipe, VLV_PCS23_DW0(ch));
-	val &= ~DPIO_LEFT_TXFIFO_RST_MASTER2;
-	val &= ~DPIO_RIGHT_TXFIFO_RST_MASTER2;
-	vlv_dpio_write(dev_priv, pipe, VLV_PCS23_DW0(ch), val);
-
 	val = vlv_dpio_read(dev_priv, pipe, VLV_PCS23_DW11(ch));
-	val &= ~DPIO_LEFT_TXFIFO_RST_MASTER;
-	val |= DPIO_RIGHT_TXFIFO_RST_MASTER;
-	val |= DPIO_LANEDESKEW_STRAP_OVRD;
+	val &= ~DPIO_LANEDESKEW_STRAP_OVRD;
 	vlv_dpio_write(dev_priv, pipe, VLV_PCS23_DW11(ch), val);
 
 	/* Deassert soft data lane reset*/
diff --git a/drivers/gpu/drm/i915/intel_hdmi.c b/drivers/gpu/drm/i915/intel_hdmi.c
index bcae111..d839db9 100644
--- a/drivers/gpu/drm/i915/intel_hdmi.c
+++ b/drivers/gpu/drm/i915/intel_hdmi.c
@@ -1494,27 +1494,13 @@ static void chv_hdmi_pre_enable(struct intel_encoder *encoder)
 
 	mutex_lock(&dev_priv->dpio_lock);
 
-	/* TX FIFO reset source */
-	val = vlv_dpio_read(dev_priv, pipe, VLV_PCS01_DW0(ch));
-	val |= DPIO_LEFT_TXFIFO_RST_MASTER2;
-	val &= ~DPIO_LEFT_TXFIFO_RST_MASTER2;
-	vlv_dpio_write(dev_priv, pipe, VLV_PCS01_DW0(ch), val);
-
+	/* allow hardware to manage TX FIFO reset source */
 	val = vlv_dpio_read(dev_priv, pipe, VLV_PCS01_DW11(ch));
-	val &= ~DPIO_LEFT_TXFIFO_RST_MASTER;
-	val &= ~DPIO_RIGHT_TXFIFO_RST_MASTER;
-	val |= DPIO_LANEDESKEW_STRAP_OVRD;
+	val &= ~DPIO_LANEDESKEW_STRAP_OVRD;
 	vlv_dpio_write(dev_priv, pipe, VLV_PCS01_DW11(ch), val);
 
-	val = vlv_dpio_read(dev_priv, pipe, VLV_PCS23_DW0(ch));
-	val &= ~DPIO_LEFT_TXFIFO_RST_MASTER2;
-	val &= ~DPIO_RIGHT_TXFIFO_RST_MASTER2;
-	vlv_dpio_write(dev_priv, pipe, VLV_PCS23_DW0(ch), val);
-
 	val = vlv_dpio_read(dev_priv, pipe, VLV_PCS23_DW11(ch));
-	val &= ~DPIO_LEFT_TXFIFO_RST_MASTER;
-	val |= DPIO_RIGHT_TXFIFO_RST_MASTER;
-	val |= DPIO_LANEDESKEW_STRAP_OVRD;
+	val &= ~DPIO_LANEDESKEW_STRAP_OVRD;
 	vlv_dpio_write(dev_priv, pipe, VLV_PCS23_DW11(ch), val);
 
 	/* Deassert soft data lane reset*/
-- 
1.7.9.5

